src/storage/data_table.cpp

```diff
@@ -189,6 +189,8 @@ template <class RowType>
 bool DataTable::SelectIntoBuffer(transaction::TransactionContext *const txn, const TupleSlot slot,
                                  RowType *const out_buffer) const {
   TERRIER_ASSERT(out_buffer->NumColumns() > 0, "The output buffer should return at least one attribute.");
+  // This cannot be visible if it's already deallocated.
+  if (!accessor_.Allocated(slot)) return false;
```

Comment:
This wasn't a change made by me, it came in when I pulled in cmu-db master.

Reply:
What's the advantage of moving this check into the function?  `SelectIntoBuffer` is called from a couple of different code points, do they all need the check or are some of them guaranteed to already meet the requirement?